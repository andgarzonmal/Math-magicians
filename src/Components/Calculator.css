.calculator-container {
  width: 85%;
  height: 50%;
  border: 1px solid black;
}

.buttons-container {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(5, 1fr);
  height: 85%;
}

.calculator-screen {
  height: 15%;
  background-color: #858694;
  border: 2px solid black;
  display: flex;
  justify-content:flex-end;
  align-items: center;
}

.solution {
  padding-right: 20px;
}

.buttons {
  background-color: azure;
  font-size: 20px;
}

.cero {
  grid-area: 5/1/5/3;
}
.orange {
  background-color: #f5913e;
}

@media screen and (min-width: 768px){

  .calculator-container {
    width: 60%;
    height: 60%;
    border: 1px solid black;
  }

}

@media screen and (min-width: 1440px){

  .calculator-container {
    width: 40%;
    height: 60%;
    border: 1px solid black;
  }

}