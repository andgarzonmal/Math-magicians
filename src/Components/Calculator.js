import React from 'react';
import './Calculator.css';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <section className="calculator-container">
        <div className="calculator-screen">
          <h1 className="solution">0</h1>
        </div>
        <div className="buttons-container">
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
          <div className="buttons"><h1>content</h1></div>
        </div>
      </section>
    );
  }
}

export default Calculator;
