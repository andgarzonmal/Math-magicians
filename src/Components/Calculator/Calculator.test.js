import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { fireEvent } from '@testing-library/dom';
import Calculator from './Calculator';

describe('Calculator', () => {
  it('The default value of the calculator screen is 0', () => {
    const component = render(<Calculator />);
    const calcScreen = component.container.querySelector('h1');
    expect(calcScreen.innerHTML).toBe('0');
  });

  it('the user sums 5+5', () => {
    const component = render(<Calculator />);
    const calcScreen = component.container.querySelector('h1');
    // user clicks on button number "5"
    let button = component.getByText('5');
    fireEvent.click(button);
    expect(calcScreen.innerHTML).toBe('5');
    // user clicks on button "+"
    button = component.getByText('+');
    fireEvent.click(button);
    expect(calcScreen.innerHTML).toBe('+');
    // user clicks on button "5"
    button = component.getByText('5');
    fireEvent.click(button);
    expect(calcScreen.innerHTML).toBe('5');
    // user clicks on button "=" to see the answer
    button = component.getByText('=');
    fireEvent.click(button);
    expect(calcScreen.innerHTML).toBe('10');
  });

  it('the user tries to divide by 0', () => {
    const component = render(<Calculator />);
    const calcScreen = component.container.querySelector('h1');
    let button = component.getByText('5');
    fireEvent.click(button);
    button = component.getByText('รท');
    fireEvent.click(button);
    button = component.getByText('0');
    fireEvent.click(button);
    button = component.getByText('=');
    fireEvent.click(button);
    expect(calcScreen.innerHTML).toBe("Can't divide by 0.");
  });
});

describe('Jest Snapshot test for Calculator', () => {
  it('renders the component', () => {
    const component = render(<Calculator />);
    expect(component).toMatchSnapshot();
  });

  it('snapshot testing the calculator component', () => {
    const calculatorComponent = renderer
      .create(
        <section className="calculator-container">
          <div className="calculator-screen">
            <h1 className="solution">0</h1>
          </div>
          <div className="buttons-container">
            <button type="button" className="buttons">
              AC
            </button>
            <button type="button" className="buttons">
              +/-
            </button>
            <button type="button" className="buttons">
              %
            </button>
            <button type="button" className="buttons orange">
              รท
            </button>
            <button type="button" className="buttons ">
              7
            </button>
            <button type="button" className="buttons ">
              8
            </button>
            <button type="button" className="buttons ">
              9
            </button>
            <button type="button" className="buttons orange">
              x
            </button>
            <button type="button" className="buttons ">
              4
            </button>
            <button type="button" className="buttons 5">
              5
            </button>
            <button type="button" className="buttons ">
              6
            </button>
            <button type="button" className="buttons orange">
              -
            </button>
            <button type="button" className="buttons ">
              1
            </button>
            <button type="button" className="buttons ">
              2
            </button>
            <button type="button" className="buttons ">
              3
            </button>
            <button type="button" className="buttons orange">
              +
            </button>
            <button type="button" className="buttons cero">
              0
            </button>
            <button type="button" className="buttons ">
              .
            </button>
            <button type="button" className="buttons orange">
              =
            </button>
          </div>
        </section>,
      )
      .toJSON();
    expect(calculatorComponent).toMatchSnapshot();
  });
});
