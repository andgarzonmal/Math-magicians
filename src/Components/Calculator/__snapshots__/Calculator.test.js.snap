// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="calculator-container"
      >
        <div
          class="calculator-screen"
        >
          <h1
            class="solution"
          >
            0
          </h1>
        </div>
        <div
          class="buttons-container"
        >
          <button
            class="buttons"
            type="button"
          >
            AC
          </button>
          <button
            class="buttons"
            type="button"
          >
            +/-
          </button>
          <button
            class="buttons"
            type="button"
          >
            %
          </button>
          <button
            class="buttons orange"
            type="button"
          >
            รท
          </button>
          <button
            class="buttons "
            type="button"
          >
            7
          </button>
          <button
            class="buttons "
            type="button"
          >
            8
          </button>
          <button
            class="buttons "
            type="button"
          >
            9
          </button>
          <button
            class="buttons orange"
            type="button"
          >
            x
          </button>
          <button
            class="buttons "
            type="button"
          >
            4
          </button>
          <button
            class="buttons 5"
            type="button"
          >
            5
          </button>
          <button
            class="buttons "
            type="button"
          >
            6
          </button>
          <button
            class="buttons orange"
            type="button"
          >
            -
          </button>
          <button
            class="buttons "
            type="button"
          >
            1
          </button>
          <button
            class="buttons "
            type="button"
          >
            2
          </button>
          <button
            class="buttons "
            type="button"
          >
            3
          </button>
          <button
            class="buttons orange"
            type="button"
          >
            +
          </button>
          <button
            class="buttons cero"
            type="button"
          >
            0
          </button>
          <button
            class="buttons "
            type="button"
          >
            .
          </button>
          <button
            class="buttons orange"
            type="button"
          >
            =
          </button>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="calculator-container"
    >
      <div
        class="calculator-screen"
      >
        <h1
          class="solution"
        >
          0
        </h1>
      </div>
      <div
        class="buttons-container"
      >
        <button
          class="buttons"
          type="button"
        >
          AC
        </button>
        <button
          class="buttons"
          type="button"
        >
          +/-
        </button>
        <button
          class="buttons"
          type="button"
        >
          %
        </button>
        <button
          class="buttons orange"
          type="button"
        >
          รท
        </button>
        <button
          class="buttons "
          type="button"
        >
          7
        </button>
        <button
          class="buttons "
          type="button"
        >
          8
        </button>
        <button
          class="buttons "
          type="button"
        >
          9
        </button>
        <button
          class="buttons orange"
          type="button"
        >
          x
        </button>
        <button
          class="buttons "
          type="button"
        >
          4
        </button>
        <button
          class="buttons 5"
          type="button"
        >
          5
        </button>
        <button
          class="buttons "
          type="button"
        >
          6
        </button>
        <button
          class="buttons orange"
          type="button"
        >
          -
        </button>
        <button
          class="buttons "
          type="button"
        >
          1
        </button>
        <button
          class="buttons "
          type="button"
        >
          2
        </button>
        <button
          class="buttons "
          type="button"
        >
          3
        </button>
        <button
          class="buttons orange"
          type="button"
        >
          +
        </button>
        <button
          class="buttons cero"
          type="button"
        >
          0
        </button>
        <button
          class="buttons "
          type="button"
        >
          .
        </button>
        <button
          class="buttons orange"
          type="button"
        >
          =
        </button>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
